### PR TITLE
implemented register method for RegistrationService and created unit tests fot it

### PR DESCRIPTION
**Pull request by vovms**

### CODE REVIEW

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 1]
```diff
@@ -5,10 +5,52 @@
...
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new UserRegistrationException("User can`t be null");
+        }
+
+        if (user.getLogin() == null) {
```
Comment (Elena-Bruyako) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:20]: Let's create separate methods for validateLogin, validatePassword and validateAge

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+
+        userWithCorrectFields = new User();
```
Comment (Elena-Bruyako) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:28]: Don't need to set up data in this way. 
Best approach is to initialise User in each test method:
given , when , then. 
It's more readable

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+                "It must be thrown UserRegistrationException for empty login");
+
+        userWithCorrectFields.setLogin("123");
+        assertThrows(UserRegistrationException.class, () -> {
+            registrationService.register(userWithCorrectFields);
+        },
+                "It must be thrown UserRegistrationException for login"
+                        + userWithCorrectFields.getLogin());
+
+        userWithCorrectFields.setLogin("12345");
```
Comment (Elena-Bruyako) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:66]: It's redundant. You can create method with blank login and another test method with register_loginLessThanMinLength_notOk

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+        assertThrows(UserRegistrationException.class, () -> {
+            registrationService.register(userWithCorrectFields);
+        },
+                "It must be thrown UserRegistrationException for login"
+                        + userWithCorrectFields.getLogin());
+    }
+
+    @Test
+    void register_loginMinLength_Ok() {
+        userWithCorrectFields.setLogin("login6");
```
Comment (Elena-Bruyako) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:76]: You should create User expected = new User(...., 
User actual = registrationService.register(expected);
assertEquals(expected, actual);


Valid cases should be checked by assertEquals, assertTrue etc

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+                "It must be thrown UserRegistrationException for empty password");
+
+        userWithCorrectFields.setPassword("123");
+        assertThrows(UserRegistrationException.class, () -> {
+            registrationService.register(userWithCorrectFields);
+        },
+                "It must be thrown UserRegistrationException for password"
+                        + userWithCorrectFields.getPassword());
+
+        userWithCorrectFields.setPassword("13245");
```
Comment (Elena-Bruyako) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:118]: The same as login comment

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+            registrationService.register(user);
+        },
+                "It must be thrown UserRegistrationException for empty password");
+    }
+
+    @Test
+    void register_passwordLessThanSix_notOk() {
+        User user = new User();
+        user.setLogin("thisLoginIsFit");
+        user.setPassword("123");
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:120]: test pre-edge case - length of 5

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+        User expected = new User();
+        expected.setLogin("thisLoginIsFit");
+        expected.setPassword("thisPasswordIsFit");
+        expected.setAge(50);
+        User actual = registrationService.register(expected);
+        assertTrue(Storage.people.contains(actual));
+    }
+
+    @Test
+    void register_nonFitUserWasNotAddedToStorage_notOk() {
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:250]: Fit - unclear terminology, did you mean `valid`?

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+
+    @Test
+    void register_maxValueAge_ok() {
+        User expected = new User();
+        expected.setLogin("thisLoginIsFit");
+        expected.setPassword("thisPasswordIsFit");
+        expected.setAge(Integer.MAX_VALUE);
+        User actual = registrationService.register(expected);
+        assertEquals(expected, actual);
+    }
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:204]: it doesn't make sense
we don't have any logic that would handle max age (because the task doesn't require it), and it's unreasonable to process such value for age

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+
+    @Test
+    void register_loginIsFit_Ok() {
+        User expected = new User();
+        expected.setPassword("thisPasswordIsFit");
+        expected.setAge(21);
+        expected.setLogin("login@1454sd");
+        User actual = registrationService.register(expected);
+        assertEquals(expected, actual);
+    }
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:90]: basically one _ok test is enough for this class. Pass the user which has all valid fields, make necessary assertions

**Review by okuzan — CHANGES_REQUESTED** [review]
Very nice job to be honest! 😎  Minor remarks:

**Review by MAMentorFX — APPROVED** [review]
